torch/csrc/jit/passes/onnx/constant_map.cpp

```diff
@@ -0,0 +1,109 @@
+#include <torch/csrc/jit/passes/onnx/constant_map.h>
+
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/passes/onnx/helper.h>
+#include <string>
+#include <iostream>
+#include <sstream>
+
+namespace torch {
+namespace jit {
+
+namespace onnx {
+using namespace ::c10::onnx;
+}
+
+std::mutex tensorShapeMutex;
+std::mutex constantTensorMutex;
+
+class TensorShapeMap;
+class ConstantTensorMap;
+
+TensorShapeMap* TensorShapeMap::instance= nullptr;
+ConstantTensorMap* ConstantTensorMap::instance= nullptr;
+
+TensorShapeMap& TensorShapeMap::getInstance(){
```

Comment:
Since we are using C++ 14, we can leverage 'static'(e.g., Meyers Singleton) to avoid such lock mechanism to speed up this a little.